napalm_nxos/nxos.py

```diff
@@ -467,17 +467,27 @@ def get_ntp_stats(self):
 
     def get_interfaces_ip(self):
 
-        interfaces_ip = dict()
-
-        command_ipv4 = 'show ip interface'
+        def get_list_of_dicts(hsh, key):
+            # Helper, lookup key in hsh, return list.
+            result = hsh.get(key, {})
+            if type(result) is dict:
+                result = [result]
+            return result
+
+        def get_interfaces_data(command):
```

Comment:
Tried adding a docstring here, but the docstring ended up being longer and more confusing than the method itself.  Thoughts?